File: 1823-determine-if-string-halves-are-alike/1823-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool isVowel(char c) {
        switch (tolower(c)) {
            case 'a':
            case 'e':
            case 'i':
            case 'o':
            case 'u':
                return true;
            default:
                return false;
        }
    }
    bool halvesAreAlike(string s) {
        int leftHalfCount = 0;
        int rightHalfCount = 0;
        //const len = s.size();
        for (int i = 0; i < s.size(); ++i) {
            if (isVowel(s[i])) {
                if (i <= (s.size() - 1)/ 2) {
                    ++leftHalfCount;
                }
                if (i >= s.size() / 2) {
                    ++rightHalfCount;
                }
            }
        }

        return leftHalfCount == rightHalfCount;
    }
};

File: 1823-determine-if-string-halves-are-alike/1823-determine-if-string-halves-are-alike.rs
impl Solution {
    pub fn is_vowel(c: char) -> bool {
        match (c) {
            'a' | 'e' | 'o' | 'i' | 'u' | 'A' | 'E' | 'O' | 'I' | 'U' => true,
            _ => false
        }
    }

    pub fn halves_are_alike(s: String) -> bool {
        let mut leftHalf = 0;
        let mut rightHalf = 0;
        let len = s.len();
        for (i, c) in s.chars().enumerate() {
            if Solution::is_vowel(c) {
                if i <= (len - 1) / 2 {
                    leftHalf += 1;
                }
                if i >= len / 2 {
                    rightHalf += 1;
                }
            }
        }

        leftHalf == rightHalf
    }
}